e2e/cephfs.go

```diff
@@ -440,6 +441,100 @@ var _ = Describe("cephfs", func() {
 				}
 			})
 
+			By("check snapshot retention", func() {
+				pvcSize := "2Gi"
+				var (
+					pvc *v1.PersistentVolumeClaim
+					app *v1.Pod
+					err error
+					// checkSum stores the md5sum of a file to verify uniqueness.
+					// checkSum string
+				)
+				createCephFSSnapshotClass(f)
+				pvc, err = loadPVC(pvcPath)
+				if pvc == nil {
+					Fail(err.Error())
+				}
+
+				pvc.Namespace = f.UniqueName
+				pvc.Spec.Resources.Requests[v1.ResourceStorage] = resource.MustParse(pvcSize)
+				e2elog.Logf("The PVC  template %+v", pvc)
+
+				app, err = loadApp(appPath)
+				if err != nil {
+					Fail(err.Error())
+				}
+				app.Namespace = f.UniqueName
+				app.Labels = map[string]string{"app": "cephfs-upgrade-testing"}
+				// create a pvc and bind it to an app.
+				err = createPVCAndApp("", f, pvc, app, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+
+				// Create snapshot of the pvc
+				snapshotPath := cephfsExamplePath + "snapshot.yaml"
+				snap := getSnapshot(snapshotPath)
+				snap.Name = "cephfs-pvc-snapshot"
+				snap.Namespace = f.UniqueName
+				snap.Spec.Source.PersistentVolumeClaimName = &pvc.Name
+				err = createSnapshot(&snap, deployTimeout)
+				if err != nil {
+					e2elog.Logf("failed to create snapshot %v", err)
+					Fail(err.Error())
+				}
+
+				// delete the parent pvc and pod
+				err = deletePVCAndApp("", f, pvc, app)
+				if err != nil {
+					Fail(err.Error())
+				}
+
+				v, err := f.ClientSet.Discovery().ServerVersion()
+				if err != nil {
+					e2elog.Logf("failed to get server version with error %v", err)
+					Fail(err.Error())
+				}
+				// pvc clone is only supported from v1.16+
+				if v.Major > "1" || (v.Major == "1" && v.Minor >= "17") {
+					pvcClone, err := loadPVC(pvcClonePath)
+					if err != nil {
+						Fail(err.Error())
+					}
+					pvcClone.Namespace = f.UniqueName
```

Comment:
where you are setting the snapshot name on pvcClone object?

Reply:
Chaning name was not required as the snapshot name set [here](https://github.com/ceph/ceph-csi/pull/1458/files/a07b47b5fe5ecc4e887d0703860a1a8bfca51403#diff-4c5b6240068e5c89f188890303acf08fR478)  it is the same mentioned in it's YAML.

Reply:
ah okay you are making use of the same name :+1: